client/fed_test.go

```diff
@@ -663,3 +665,296 @@ func TestNewTransferJob(t *testing.T) {
+
+			// Create a pipe for output
+			r, w, _ := os.Pipe()
+			originalStdout := os.Stdout
+			os.Stdout = w
+
+			go func() {
+				if export.Capabilities.PublicReads {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithDirOnlyOption(true))
+					require.NoError(t, err)
+					w.Close()
+				} else {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithDirOnlyOption(true), client.WithTokenLocation(tempToken.Name()))
+					require.NoError(t, err)
+					w.Close()
+				}
+			}()
+			var buf bytes.Buffer
+			_, err = io.Copy(&buf, r)
+			require.NoError(t, err)
+			os.Stdout = originalStdout
+
+			assert.Contains(t, buf.String(), "test")
+			assert.NotContains(t, buf.String(), "hello_world.txt")
+		}
+	})
+
+	// Test that fileonly works and only lists files
+	t.Run("testPelicanObjectLsFileOnly", func(t *testing.T) {
+		// Set path for object to upload/download
+		for _, export := range fed.Exports {
+			uploadURL := fmt.Sprintf("pelican://%s:%s%s", param.Server_Hostname.GetString(), strconv.Itoa(param.Server_WebPort.GetInt()), export.FederationPrefix)
+
+			// Create a pipe for output
+			r, w, _ := os.Pipe()
+			originalStdout := os.Stdout
+			os.Stdout = w
+
+			go func() {
+				if export.Capabilities.PublicReads {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithFileOnlyOption(true))
+					require.NoError(t, err)
+					w.Close()
+				} else {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithFileOnlyOption(true), client.WithTokenLocation(tempToken.Name()))
+					require.NoError(t, err)
+					w.Close()
+				}
+			}()
+			var buf bytes.Buffer
+			_, err = io.Copy(&buf, r)
+			require.NoError(t, err)
+			os.Stdout = originalStdout
+
+			assert.NotContains(t, buf.String(), "test")
+			assert.Contains(t, buf.String(), "hello_world.txt")
+		}
+	})
+
+	// Test that long works
+	t.Run("testPelicanObjectLsLong", func(t *testing.T) {
+		// Set path for object to upload/download
+		for _, export := range fed.Exports {
+			uploadURL := fmt.Sprintf("pelican://%s:%s%s", param.Server_Hostname.GetString(), strconv.Itoa(param.Server_WebPort.GetInt()), export.FederationPrefix)
+
+			// Create a pipe for output
+			r, w, _ := os.Pipe()
+			originalStdout := os.Stdout
+			os.Stdout = w
+
+			go func() {
+				if export.Capabilities.PublicReads {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithLongOption(true))
+					require.NoError(t, err)
+					w.Close()
+				} else {
+					err := client.DoList(fed.Ctx, uploadURL, client.WithLongOption(true), client.WithTokenLocation(tempToken.Name()))
+					require.NoError(t, err)
+					w.Close()
+				}
+			}()
+			var buf bytes.Buffer
+			_, err = io.Copy(&buf, r)
+			require.NoError(t, err)
+			os.Stdout = originalStdout
+
+			assert.Contains(t, buf.String(), "test")
+			// permissions for test
+			assert.Contains(t, buf.String(), "drwxrwxr-x")
```

Comment:
@bbockelm if we're working on `ls` and `stat` in a way that starts reporting directory and permissions information like this, are we moving away from the object framework and toward something closer to a filesystem? Some of this doesn't make much sense to me unless we stop saying Pelican federates objects.

Reply:
That is a good question, the permissions might not make too much sense in some cases but might be useful in others. Let me know what you guys think and if these permissions make sense as something to be printed out. I can easily remove this if we do not want this feature